### PR TITLE
feature/M2-738: added audio player item

### PR DESCRIPTION
**Pull request by alex-i-e**

### CODE REVIEW

**Comment by alex-i-e** [comment]
image
image

File: src/shared/components/MediaUploader/MediaUploader.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import { Trans, useTranslation } from 'react-i18next';
+
+import { StyledBodyMedium, theme, variables } from 'shared/styles';
+import { Svg, Tooltip } from 'shared/components/index';
```
Comment (kdanilenka32) [inline src/shared/components/MediaUploader/MediaUploader.tsx:4]: shared/components

File: src/shared/components/MediaUploader/MediaUploader.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import {
+  StyledContainer,
+  StyledNameWrapper,
+  StyledSourceContainer,
+  StyledSvg,
+  StyledTitle,
+} from 'shared/components/MediaUploader/MediaUploader.styles';
+import { useMediaUploader } from 'shared/components/MediaUploader/MediaUploader.hooks';
+import { MediaUploaderProps } from 'shared/components/MediaUploader/MediaUploader.types';
+import { MLPlayer } from 'shared/components/MLPlayer';
```
Comment (kdanilenka32) [inline src/shared/components/MediaUploader/MediaUploader.tsx:15]: pls use relative import

File: src/shared/components/MediaUploader/MediaUploader.styles.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+import {
+  StyledFlexAllCenter,
+  StyledFlexColumn,
+  StyledTitleMedium,
+  theme,
+  variables,
+} from 'shared/styles';
+import { shouldForwardProp } from 'shared/utils/shouldForwardProp';
+import { Svg } from 'shared/components';
```
Comment (kdanilenka32) [inline src/shared/components/MediaUploader/MediaUploader.styles.ts:12]: shared/utils

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayer.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  StyledClearedButton,
+  StyledFlexTopCenter,
+  StyledLabelBoldLarge,
+  StyledTitleMedium,
+  theme,
+  variables,
+} from 'shared/styles';
+import { Svg } from 'shared/components';
+import { InputController } from 'shared/components/FormComponents';
+import { ItemInputTypes } from 'shared/types/activityItems';
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayer.tsx:16]: pls use shared/types

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayer.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+import { MediaUploader } from 'shared/components/MediaUploader';
+import { ResourceDataType } from 'shared/components/MediaUploader/MediaUploader.types';
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayer.tsx:18]: pls use shared/components

File: src/shared/components/MediaUploader/MediaUploader.hooks.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
+import { ChangeEvent, DragEvent, MouseEvent, useRef, useState } from 'react';
+
+import { ResourceDataType } from './MediaUploader.types';
+
+const MAX_FILE_SIZE = 8388608; // 8 MB
+
+export type MediaUploaderProps = {
+  setResourceData: (data: ResourceDataType | null) => void;
```
Comment (kdanilenka32) [inline src/shared/components/MediaUploader/MediaUploader.hooks.ts:8]: pls move to types

File: src/shared/components/MLPlayer/MLPlayer.styles.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+`;
+
+export const StyledNameWrapper = styled(StyledFlexTopCenter)`
+  font-size: ${variables.font.size.md};
+  color: ${variables.palette.on_surface_variant};
+  display: flex;
+  position: absolute;
+  top: 0;
+
+  && {
```
Comment (kdanilenka32) [inline src/shared/components/MLPlayer/MLPlayer.styles.ts:39]: do y need the  && {?

File: src/shared/components/MLPlayer/MLPlayer.hooks.ts [hunk 8]
```diff
@@ -0,0 +1,120 @@
+import { SyntheticEvent, useEffect, useMemo, useRef, useState } from 'react';
+import { OnProgressProps } from 'react-player/base';
+import ReactPlayer from 'react-player';
+
+import { ResourceDataType } from 'shared/components/MediaUploader/MediaUploader.types';
```
Comment (kdanilenka32) [inline src/shared/components/MLPlayer/MLPlayer.hooks.ts:5]: pls use abs path

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayer.tsx (hunk 5 above)
Comment (kdanilenka32) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayer.tsx:18]: pls change imports

File: src/modules/Builder/features/ActivityItems/ItemConfiguration/ItemConfiguration.types.ts [hunk 9]
```diff
@@ -63,6 +63,9 @@ export type ItemConfigurationForm = {
   sliderOptions?: SliderOption[];
   textResponseAnswer?: string;
   textResponseMaxCharacters?: number;
+
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivityItems/ItemConfiguration/ItemConfiguration.types.ts:66]: pls remove spacing

File: src/shared/components/MLPlayer/MLPlayer.hooks.ts [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+
+import { MLPlayerStateProps } from './MLPlayer.types';
+
+const calculateTime = (secs: number) => {
+  const minutes = Math.floor(secs / 60);
+  const seconds = Math.floor(secs % 60);
+  const returnedSeconds = seconds < 10 ? `0${seconds}` : `${seconds}`;
```
Comment (IhorNesterenko99) [inline src/shared/components/MLPlayer/MLPlayer.hooks.ts:12]: seconds < 10 ? `0${seconds}` : seconds;

File: src/shared/components/MLPlayer/MLPlayer.hooks.ts
Comment (IhorNesterenko99) [inline src/shared/components/MLPlayer/MLPlayer.hooks.ts:1]: Do we need MLPlayer and MediaUploader in shared components?
I think it's better to place them in Builder components

File: src/shared/components/MLPlayer/MLPlayer.styles.ts (hunk 7 above)
Comment (alex-i-e) [reply src/shared/components/MLPlayer/MLPlayer.styles.ts:39]: fixed

File: src/shared/components/MLPlayer/MLPlayer.hooks.ts (hunk 10 above)
Comment (alex-i-e) [reply src/shared/components/MLPlayer/MLPlayer.hooks.ts:12]: for ex.: 
9 sec => 0:09
10 sec => 0:10
69 sec => 1:09
70 sec => 1:10

File: src/shared/components/MLPlayer/MLPlayer.hooks.ts
Comment (alex-i-e) [reply src/shared/components/MLPlayer/MLPlayer.hooks.ts:1]: agree

**Review by IhorNesterenko99 — APPROVED** [review]

**Review by kdanilenka32 — APPROVED** [review]

**Pull request merged**